python/pyspark/sql/streaming/state_api_client.py

```diff
@@ -0,0 +1,142 @@
+        print(f"setting handle state to: {state}")
+        proto_state = self._get_proto_state(state)
+        set_handle_state = stateMessage.SetHandleState(state=proto_state)
+        handle_call = stateMessage.StatefulProcessorCall(setHandleState=set_handle_state)
+        message = stateMessage.StateRequest(statefulProcessorCall=handle_call)
+
+        self._send_proto_message(message)
+
+        response_message = self._receive_proto_message()
+        status = response_message.statusCode
+        if (status == 0):
+            self.handle_state = state
+        else:
+            raise Exception(f"Error setting handle state: {response_message.errorMessage}")
+        print(f"setHandleState status= {status}")
+
+    def set_implicit_key(self, key: str) -> None:
+        print(f"setting implicit key: {key}")
+        set_implicit_key = stateMessage.SetImplicitKey(key=key)
+        request = stateMessage.ImplicitGroupingKeyRequest(setImplicitKey=set_implicit_key)
+        message = stateMessage.StateRequest(implicitGroupingKeyRequest=request)
+
+        self._send_proto_message(message)
+        response_message = self._receive_proto_message()
+        status = response_message.statusCode
+        print(f"setImplicitKey status= {status}")
+        if (status != 0):
+            raise Exception(f"Error setting implicit key: {response_message.errorMessage}")
+
+    def remove_implicit_key(self) -> None:
+        print(f"removing implicit key")
+        remove_implicit_key = stateMessage.RemoveImplicitKey()
+        request = stateMessage.ImplicitGroupingKeyRequest(removeImplicitKey=remove_implicit_key)
+        message = stateMessage.StateRequest(implicitGroupingKeyRequest=request)
+
+        self._send_proto_message(message)
+        response_message = self._receive_proto_message()
+        status = response_message.statusCode
+        print(f"removeImplicitKey status= {status}")
+        if (status != 0):
+            raise Exception(f"Error removing implicit key: {response_message.errorMessage}")
+
+    def get_value_state(self, state_name: str, schema: Union[StructType, str]) -> None:
+        if isinstance(schema, str):
+            schema = cast(StructType, _parse_datatype_string(schema))
+
+        print(f"initializing value state: {state_name}")
+
+        state_call_command = stateMessage.StateCallCommand()
+        state_call_command.stateName = state_name
+        state_call_command.schema = schema.json()
+        call = stateMessage.StatefulProcessorCall(getValueState=state_call_command)
+        message = stateMessage.StateRequest(statefulProcessorCall=call)
+
+        self._send_proto_message(message)
+        response_message = self._receive_proto_message()
+        status = response_message.statusCode
+        if (status != 0):
+            raise Exception(f"Error initializing value state: {response_message.errorMessage}")
+        print(f"getValueState status= {status}")
+
+    def _get_proto_state(self,
```

Comment:
Is double underscore for starting chars the recommended practice here for private methods ?

cc - @HyukjinKwon to confirm

Reply:
yes if this is meant to be private